skl/skl.go

```diff
@@ -341,13 +343,13 @@ func (s *Skiplist) PutWithHint(key []byte, v y.ValueStruct, hint *Hint) {
 				recomputeHeight++
 			} else if hint.prev[recomputeHeight] != s.head &&
 				hint.prev[recomputeHeight] != nil &&
-				bytes.Compare(key, hint.prev[recomputeHeight].key(s.arena)) <= 0 {
+				y.CompareKeys(key, hint.prev[recomputeHeight].key(s.arena)) <= 0 {
```

Comment:
Why change this?

Reply:
`Skl` use `y.CompareKeys` in other method such as `findNear`, `findSpliceForLevel` etc...
Use `bytes.Compare` here is incorrect, actually it cause a lots of `ErrKeyNotFound` during test. (I just notice it in this PR)